taglib-core/src/test/java/com/reevoo/client/ReevooMarkClientTest.java

```diff
@@ -220,4 +228,80 @@ public void ifResponse200ReturnResponseBodyIndependentlyOfTheNumberOfReviews() t
 
     }
 
+    @Test
+    public void testFailedRequestAreCachedByConfiguredNumberOfSeconds() throws IOException {
+        when(m.getStatusCode()).thenReturn(500);
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.SECOND, 60);
+        Date expected_time = cal.getTime();
+
+        c.obtainReevooMarkData(m);
+        Date cache_expires = Cache.get("http://www.example.org/reevoomark").getExpirationTime();
+
+        assertEquals((double) expected_time.getTime(), (double) cache_expires.getTime(), 1000);
+    }
+
+    @Test
+    public void testConsecutiveFailedRequestsIncrementFailedCounter() throws IOException {
+        when(m.getStatusCode()).thenReturn(404);
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.SECOND, -10000);
+        for (int i = 0; i < 4; i++) {
+            c.obtainReevooMarkData(m);
+            Cache.get("http://www.example.org/reevoomark").setExpirationTime(cal.getTime());
+        }
+        assertEquals(4, Cache.get("http://www.example.org/reevoomark").getConsecutiveFailedAttempts());
+    }
+
+    @Test
+    public void testConsecutiveFailedRequestsOverLimitTriggerCircuitBreaker() throws IOException {
+        when(m.getStatusCode()).thenReturn(404);
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.SECOND, -10000);
```

Comment:
I would extract these lines that are common between this tests and put it in a expireCache() method